proto/ibc/core/port/v1/query.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package ibc.core.port.v1;
+
+option go_package = "github.com/cosmos/ibc-go/modules/core/05-port/types";
+
+import "ibc/core/channel/v1/channel.proto";
+
+// Query provides defines the gRPC querier service
+service Query {
+  // Port queries an IBC Port.
+  rpc Port(QueryPortRequest) returns (QueryPortResponse) {}
```

Comment:
Perhaps the naming here is too generic and would be better suited as something more applicable/specific to port versions. 

Suggestions welcome!

Reply:
Another open question here is: should a HTTP interface using grpc gateway sit in front of this endpoint as well?

Reply:
wdyt about `PortVersion`?

Reply:
I imagine I'm querying for a resource which is in this case is the `CounterpartyVersion` given a port-id

Reply:
So this query will be invoked by a relayer on the host chain, post `OnChanOpenInit` on the controller and pre `OnChanOpenTry` on the host, correct? 
We are querying for the `CounterpartyVersion` as from the perspective of the controller, but the chain servicing the request is the counterparty so maybe `PortVersion` could be a better fit. 

Correct me if I'm wrong here! Would be interested to hear @colin-axner's thoughts also on naming conventions

Reply:
What about just `AppVersionNegotiation`? This is supposed to be called pre OnChanOpenTry, but really it is just application version negotiation based on some suggested counterparty version

Reply:
Kinda nitpicking here but I guess it depends on how we want to structure our overall API. My instinct was to have our query endpoints return a particular resource (and thus have the naming follow that), but seeing as this is not really returning anything from state maybe that doesn't fit. 

I'm happy with whatever you both decide.

Reply:
Nitpicking is good because it is harder to change this later. What about `AppVersion`?

Reply:
Indeed! Naming can be tricky to get right, and it's 100% harder to change after the fact.

Currently named as `NegotiateAppVersion`, I'd be for shorting this to `AppVersion` if you both agree, and request/response would be `QueryAppVersionRequest` and `QueryAppVersionResponse` respectively. 
Does `ChannelVersion` fit the use case here? If so I'd be open to it as an alternative but my instincts are favouring one of `NegotiateAppVersion` or just `AppVersion`

Reply:
Lets go with `AppVersion` if @seantking agrees. `AppVersion` is the version stored in the channel, so sometimes it is referred to as channel version. It isn't technically referenced as `AppVersion` anywhere (yet), but it is the version for the application so I think it makes sense to head in this direction